### PR TITLE
fixing broken linkpost url parsing

### PR DESCRIPTION
**Pull request by b0b3rt**
Fixing an issue introduced in https://github.com/ForumMagnum/ForumMagnum/pull/9597, where a linkpost url without the url protocol (i.e. `https`) fails to get parsed by `new URL` and blows up the entire post page.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208328020782978) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx [hunk 1]
```diff
@@ -126,12 +126,16 @@ const LWPostsPageHeader = ({post, showEmbeddedPlayer, toggleEmbeddedPlayer, clas
   answerCount?: number,
 }) => {
   const { PostsPageTitle, PostsAuthors, LWTooltip, PostsPageDate, CrosspostHeaderIcon, PostsGroupDetails, PostsTopSequencesNav, PostsPageEventData, AddToCalendarButton, GroupLinks, LWPostsPageHeaderTopRight, PostsAudioPlayerWrapper, PostsVote, AudioToggle, PostActionsButton, AlignmentCrosspostLink, ReadTime, LWCommentCount } = Components;
-  // eslint-disable-next-line react-hooks/exhaustive-deps
 
   const rssFeedSource = ('feed' in post) ? post.feed : null;
-  const feedLinkDescription = rssFeedSource?.url && getHostname(rssFeedSource.url)
-  const feedLink = rssFeedSource?.url && `${getProtocol(rssFeedSource.url)}//${getHostname(rssFeedSource.url)}`;
-  const feedDomain = feedLink && new URL(feedLink).hostname;
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx:134]: I think this stuff was copied from `PostsPagePostHeader` - can you make the same updates to that component as well?

**Review by s-cheng — APPROVED** [review]
LGTM - I didn't test this since it's in a LW component, I will assume you tested it :)

File: packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx [hunk 2]
```diff
@@ -126,12 +126,16 @@ const LWPostsPageHeader = ({post, showEmbeddedPlayer, toggleEmbeddedPlayer, clas
...
+  let feedLinkDomain;
+  let feedLink;
+  if (rssFeedSource?.url) {
+    let feedLinkProtocol;
+    ({ hostname: feedLinkDomain, protocol: feedLinkProtocol } = parseUnsafeUrl(rssFeedSource.url));
+    feedLink = `${feedLinkProtocol}//${feedLinkDomain}`;
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx:136]: Actually is this missing a `:`?

Comment (b0b3rt) [reply packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx:136]: Empirically I think no, but will double-check

Comment (b0b3rt) [reply packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx:136]: No, it looks like `URL.protocol` returns with a semicolon at the end

File: packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx (hunk 1 above)
Comment (b0b3rt) [reply packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx:134]: Good catch, thanks!

File: packages/lesswrong/components/posts/PostsPage/PostsPagePostHeader.tsx [hunk 3]
```diff
@@ -164,6 +165,24 @@ export function getHostname(url: string): string {
   return parser.hostname;
 }
 
+export function parseUnsafeUrl(url: string) {
```
[the remark this answers is not in the document]
Comment (b0b3rt) [reply packages/lesswrong/components/posts/PostsPage/PostsPagePostHeader.tsx:168]: Yep, done

**Pull request merged**